cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -611,6 +621,112 @@ mkConsensusTracers mbEKGDirect trSel verb tr nodeKern fStats = do
      when (isRollForward ev) $
        sendEKGDirectCounter ekgDirect "cardano.node.metrics.served.header.counter.int"
 
+-- | CdfCounter tracks the number of time a value below 'limit' has been seen.
+newtype CdfCounter (limit :: Nat) = CdfCounter Int64
+
+-- | Estimates the CDF for a specific limit 'l' by counting the number of times
+-- a value 'v' is below the limit.
+cdfCounter :: forall a l.
+               ( Num a, Ord a
+               , KnownNat l)
+            => a -> Int -> Int64 -> STM.TVar (CdfCounter l) -> STM Double
+cdfCounter v !size !step tCdf= do
+    when (v < lim) $
+        STM.modifyTVar' tCdf (\(CdfCounter c) -> CdfCounter $ c + step)
+
+    (CdfCounter cdf) <- STM.readTVar tCdf
+    return $! (fromIntegral cdf / fromIntegral size)
+
+  where
+    lim :: a
+    lim = fromInteger $ natVal (Proxy :: Proxy l)
+
+
+-- Add an observation to the CdfCounter.
+incCdfCounter :: Ord a => Num a => KnownNat l => a -> Int -> STM.TVar (CdfCounter l) -> STM Double
+incCdfCounter v size = cdfCounter v size 1
+
+-- Remove an observation from the CdfCounter.
+decCdfCounter :: Ord a => Num a => KnownNat l => a -> Int -> STM.TVar (CdfCounter l) -> STM Double
+decCdfCounter v size = cdfCounter v size (-1)
+
+traceBlockFetchClientMetrics
+  :: forall blk remotePeer.
+     ( )
+  => Maybe EKGDirect
+  -> STM.TVar (IntPSQ SlotNo NominalDiffTime)
+  -> STM.TVar (CdfCounter 1)
+  -> STM.TVar (CdfCounter 2)
+  -> STM.TVar (CdfCounter 5)
+  -> Tracer IO (TraceLabelPeer remotePeer (TraceFetchClientState (Header blk)))
+  -> Tracer IO (TraceLabelPeer remotePeer (TraceFetchClientState (Header blk)))
+traceBlockFetchClientMetrics Nothing _ _ _ _ tracer = tracer
+traceBlockFetchClientMetrics (Just ekgDirect) slotMapVar cdf1sVar cdf2sVar cdf5sVar tracer = Tracer bfTracer
+
+  where
+    incCdfs :: NominalDiffTime -> Int -> STM (Double, Double, Double)
+    incCdfs delay size = do
+      cdf1s <- incCdfCounter delay size cdf1sVar
+      cdf2s <- incCdfCounter delay size cdf2sVar
+      cdf5s <- incCdfCounter delay size cdf5sVar
+      return (cdf1s, cdf2s, cdf5s)
+
+    decCdfs :: NominalDiffTime -> Int -> STM ()
+    decCdfs delay size =
+      decCdfCounter delay size cdf1sVar
```

Comment:
nicer indentation:
```suggestion
         decCdfCounter delay size cdf1sVar
```

Reply:
Rest of the file uses two space indentation.